api/src/services/osmesa.js

```diff
@@ -1,7 +1,15 @@
+const pg = require('pg')
+// conversion of postgresql bigint (int8) to JS number (from https://github.com/knex/knex/issues/387)
+pg.types.setTypeParser(pg.types.builtins.INT8, 'text', parseInt)
```

Comment:
Was there a test failure that caused this? Do we need this anywhere else in the app as a setting?

Reply:
@kamicut great question! There was no test failure, but in the new `getTeamStats` method, it sums over the user statistics, so needs to parse strings into integers for some columns. This is one possible workaround, which seemed cleanest to me. I got it from the linked github issue in the knex repo.

### Background

This workaround is needed because any PostgreSQL  bigint column will be returned by Knex as a string, because the bigint values could be greater than JavaScript `Number.MAX_VALUE`. 

I think a safe bet is we are not going to have values larger than 2<sup>1024</sup> coming from this particular query. If we think it's possible for that to happen, then it is likely the whole app, not just this module, will need to be refactored to use a math library that supports large numbers, like the new BigInt feature, or some other existing math library.

> Do we need this anywhere else in the app as a setting?

The greater question is how are we dealing with very large numbers in the app overall? It's possible in other places in the app that bigint queries are being returned and strings and are being handled differently, like just displayed as raw string values. 

So one option is making a bigger scoped issue doing the due diligence about handling large numbers coming from PostgreSQL. 

But I am also comfortable just scoping this change to this module, and not automatically  doing it everywhere knex is querying PostgreSQL. I assume this  `setTypeParser` call only effects this module, because it did not cause any test regressions elsewhere in the app. But I do not actually know that 100% certainty.

Reply:
I don't think we have values anywhere near that resolution unless users start mapping the entire world at the picometre (10^-12) scale 😂. It might be however useful to centralize string to number conversions somewhere. Let's discuss this on a call.